spec/models/user_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+
+  describe "validations" do
+    it { should validate_presence_of :email }
+    it { should validate_uniqueness_of :email }
+    it { should validate_presence_of :role }
+    it { should define_enum_for :role }
+    it { should validate_presence_of :password}
+    it { should have_secure_password }
+  end
+
+  describe "user roles" do
+    it "user can be student user" do
+      data = {email: "sample@email.com", password: "1234", role: 0}
+      student_user = User.create(data)
+      expect(student_user).to be_valid
+      expect(student_user.role).to eq("student")
+      expect(student_user.student?).to be_truthy
```

Comment:
Very cool. I did not know be_truthy was available in capybara 🙂

Reply:
capybara is a very powerful tool for sure 😄